File: LeetSync/1386-shift-2d-grid/shift-2d-grid.py
from typing import List

class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        m, n = len(grid), len(grid[0])
        total = m * n
        
        k %= total
        
        result = [[0] * n for _ in range(m)]
        
        for i in range(m):
            for j in range(n):
                old_idx = i * n + j
                new_idx = (old_idx + k) % total
                
                new_i, new_j = divmod(new_idx, n)
                
                result[new_i][new_j] = grid[i][j]
                
        return result

File: LeetSync/1386-shift-2d-grid/shift-2d-grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int m = grid.size();
        int n = grid[0].size();
        int total = m * n;
        
        k %= total;
        
        vector<vector<int>> result(m, vector<int>(n));
        
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                int old_idx = i * n + j;
                int new_idx = (old_idx + k) % total;
                
                int new_i = new_idx / n;
                int new_j = new_idx % n;
                
                result[new_i][new_j] = grid[i][j];
            }
        }
        
        return result;
    }
};